great_expectations/data_context/util.py

```diff
@@ -567,3 +570,79 @@ def mask_db_url(url: str, use_urlparse: bool = False, **kwargs) -> str:
                 )
 
             return masked_url
+
+    @classmethod
+    def sanitize_data_context_config(cls, config: dict) -> dict:
```

Comment:
I think this file is great but is there a way we could possibly genericize this whole utility? I'm wary of having to constantly expand this class to fit different core GE objects. I think it's probably fine as is but what are your thoughts around a recursive method that anonymizes anything in a list of keywords?

Something like:
```python
SENSITIVE_STRINGS = set(...)

@classmethod
def sanitize(cls, config: dict) -> dict:
    if not isinstance(config, dict):
        raise

    config_copy = safe_deep_copy(config)
    
    def _recursive_cleaner_method(config: Any) -> None:
        if isinstance(config, dict):
            for key, val in config.items():
                if key in self.SENSITIVE_STRINGS and isinstance(val, str):
                    # obfuscate
                _recursive_cleaner_method(val)
        elif isinstance(config, list):
            for val in config:
                _recursive_cleaner_method(val)

    _recursive_cleaner_method(config_config) # Perform anonymization in place
    
    return config_copy
```

Reply:
thanks @cdkini, this is a great idea -- I had originally decided against this because there is different processing required depending on the key value, but i believe integrating our approaches works well.